cfme/tests/cli/test_appliance_update.py

```diff
@@ -65,18 +65,14 @@ def appliance_preupdate(old_version, appliance):
     repo file for update"""
 
     usable = []
-    sp = SproutClient.from_config()
+    sp = SproutClient.from_config(sprout_user_key=pytest.config.option.sprout_user_key or None)
```

Comment:
I think it would be better to actually check this first. If no sprout_user_key is passed, then this ends up failing because it requires auth.

You can do something like this to make it behave in a more predictable way:
```
sprout_key = pytest.config.option.get('sprout_user_key')
if sprout_key is None:
    pytest.skip('Missing sprout auth key, can not run setup')
SproutClient.from_config(sprout_key)
```

Reply:
Not really. There is additional logic in the from_config (reading the env vars)

Reply:
I also don't understand why SKIP is better then failing the test with ERROR. I have better info about the failure from the ERROR -- the traceback.

Reply:
Sure, not a great pattern overall then for using SproutClient.from_config.